ApnsPHP/SharedConfig.php

```diff
@@ -129,8 +129,12 @@ abstract class SharedConfig
      * @throws BaseException if the environment is not
      *         sandbox or production or the provider certificate file is not readable.
      */
-    public function __construct($environment, $providerCertificateFile, $protocol = self::PROTOCOL_BINARY)
+    public function __construct($environment, $providerCertificateFile, $protocol = self::PROTOCOL_HTTP)
     {
+        if ($protocol === self::PROTOCOL_BINARY) {
+            @trigger_error('Binary protocol is deprecated and will be removed in the next version', E_USER_DEPRECATED);
```

Comment:
This should just use the logger. There is no need to make it more than that.

Reply:
I think it's more a library change here, to tell devs that the next version this behavior will change. I often use `E_USER_DEPRECATED` in my exception handler for example and that would not be possible with log.

Reply:
But we have a logger, so we should just use that one. We can do `$this->logger->warning('stuff')` to issue a warning.